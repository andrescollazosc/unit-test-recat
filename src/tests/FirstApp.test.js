import React from "react";
import FirstApp from "../FirstApp.App";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

describe("Test <FirstApp />", () => {
  //   test('should to show a message "Hello, Andres"', () => {
  //     const greeting = "Hello, Andres";
  //     const { getByText } = render(<FirstApp greeting={greeting}/>);

  //     expect(getByText(greeting)).toBeInTheDocument();
  //   });

  test("should showe <FirstApp /> correctly", () => {
    const greeting = "Hello, Andres";
    const wrapper = shallow(<FirstApp greeting={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });
  0;
  test("should show the sutitle that was sending by props", () => {
    const greeting = "Hello, Andres";
    const subTittle = "Soy un subtitulo";
    const wrapper = shallow(
      <FirstApp greeting={greeting} subTitle={subTittle} />
    );

    const textPhra = wrapper.find('p').text();
    expect(textPhra).toBe(subTittle);
  });
});
