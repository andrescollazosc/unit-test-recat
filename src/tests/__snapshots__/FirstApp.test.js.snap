// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FirstApp /> should showe <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hello, Andres
  </h1>
  <p>
    I am a subtitle
  </p>
</Fragment>
`;
